geode-redis/src/main/java/org/apache/geode/redis/GeodeRedisServer.java

```diff
@@ -446,6 +449,7 @@ private void initializeRedis() {
         throw assErr;
       }
       this.keyRegistrar = new KeyRegistrar(redisMetaData);
+      this.pubSub = createPubSub();
```

Comment:
That is true.

Reply:
But I don't see any immediate value in changing it.

Reply:
Also, you can use `new PubSub` as the `registerFunction` can (and should) be called in the constructor of `PubSub` otherwise one is exposing knowledge that does not need to be exposed.